config/extension_hook.go

```diff
@@ -180,7 +180,7 @@ func (list HookList) Run(status error, args ...interface{}) (result []interface{
 		currentErr = shell.FilterPlanError(currentErr, hook.options().TerraformCliArgs[0])
 		if _, ok := currentErr.(errors.PlanWithChanges); ok {
 			errs = append(errs, currentErr)
-		} else if currentErr != nil {
+		} else if currentErr != nil && !hook.IgnoreError {
```

Comment:
I think the original idea was to run the hook command even if an error occurred prior the error. From what I can see in the code, that is what is implemented.

I agree that the comment or name is wrong, but that could change the behavior.

Reply:
The original goal was to run the cleanup at the end whether there is an error or not.

Reply:
Ah I see. The description isn't very clear `continue execution on error`

Reply:
The concept was first introduced on extra_command. But the meaning of this one was more execute even if there were error before...

Reply:
I think both concepts have values and this one should clearly renamed.

The error should not be ignored without leaving a trace in the logs though.

Reply:
It leaves a trace right now with this implementation

I think `ignore_error` should be what I'm trying to do and there could be a new option `run_on_errors` for the current behavior. What do you think?

Reply:
Yes precisely.